### PR TITLE
Move common Makefile targets to Makefile.common

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+		fi'
+endef
+
+TYPHA_BRANCH?=$(PIN_BRANCH)
+TYPHA_REPO?=github.com/projectcalico/typha
+LIBCALICO_BRANCH?=$(PIN_BRANCH)
+LIBCALICO_REPO?=github.com/projectcalico/libcalico-go
+CONFD_BRANCH?=$(PIN_BRANCH)
+CONFD_REPO?=github.com/projectcalico/confd
+FELIX_BRANCH?=$(PIN_BRANCH)
```
Comment (tmjd) [inline Makefile.common]: Should we have pin update targets for CONFD and FELIX here also?

Comment (rafaelvanoni) [reply Makefile.common]: See https://github.com/projectcalico/felix/pull/2178 for an example use case.

**Comment by tmjd** [comment]
Did you consider adding this in a different repo (or even a new one)?
I feel like this doesn't really fit here. What about having it in libcalico-go? Or we could make a new repo but that seems a little silly, but I could be convinced.
Has this discussion already been had and I'm late to the party?

**Comment by rafaelvanoni** [comment]
> Did you consider adding this in a different repo (or even a new one)?
> I feel like this doesn't really fit here. What about having it in libcalico-go? Or we could make a new repo but that seems a little silly, but I could be convinced.
> Has this discussion already been had and I'm late to the party?

There was a short discussion in slack of where to put it. My original idea was to have this in its own repo, but Shaun pointed out we need to a better versioning scheme than branches/releases. So I thought of moving it here and version it along with GO_BUILD_VER.

File: Makefile.common (hunk 1 above)
Comment (tmjd) [reply Makefile.common]: Thanks for the example. Are you going to add `update-confd-pin` and `update-felix-pin` targets?

File: Makefile.common [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+endif
+
+# For building, we use the go-build image for the *host* architecture, even if the target is different
+# the one for the host should contain all the necessary cross-compilation tools
+# we do not need to use the arch since go-build:v0.15 now is multi-arch manifest
+CALICO_BUILD=calico/go-build:$(GO_BUILD_VER)
+ETCD_VERSION?=v3.3.7
+K8S_VERSION?=v1.14.1
+PROTOC_VER?=v0.1
+PROTOC_CONTAINER ?=calico/protoc:$(PROTOC_VER)-$(BUILDARCH)
```
Comment (tmjd) [inline Makefile.common]: The code here is correct but in the v0.27 version these 2 lines are mixed up.

**Comment by rafaelvanoni** [comment]
Already fixed

**Pull request closed without merging**